src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -195,6 +197,60 @@ public TutorialClass findTutorialClassFromList(TutorialClass tutorialClass, Modu
         return tutorialClassInList;
     }
 
+    /**
+     * Randomly allocates the students in {@code tutorial class} into {@code numOfTeams} of different teams.
+     *
+     * @param moduleCode of the tutorial class.
+     * @param tutorialClass to allocate the different students into the teams to.
+     * @param numOfTeams of teams to split into.
+     */
+    public void randomTeamAllocation(ModuleCode moduleCode, TutorialClass tutorialClass, int numOfTeams) {
+        requireAllNonNull(moduleCode, tutorialClass, numOfTeams);
+        ArrayList<TutorialTeam> teams = tutorialClass.getTeams();
+        ArrayList<Person> classList = tutorialClass.getStudents();
+        int classSize = tutorialClass.getStudents().size();
+        int teamSize = (int) Math.ceil((double) classSize / numOfTeams);
+        teams.clear();
+
+        // creating the teams to add into
+        for (int i = 1; i <= numOfTeams; i++) {
+            String teamName = "Team" + i;
+            TutorialTeam team = new TutorialTeam(teamName, teamSize);
+            addTeam(tutorialClass, team);
+        }
+
+        Random random = new Random();
+        int k = 0;
+        while (k < classList.size()) {
+            int randInt = random.nextInt(numOfTeams);
+            TutorialTeam currTeam = tutorialClass.getTeams().get(randInt);
+            if (hasTeamSizeExceeded(currTeam)) {
+                continue;
+            }
```

Comment:
Based on my understanding, if the chosen random team is full, it will continue and skip to the next iteration. However, since the loop iterates over every student, would this mean that it is possible that some students would not get assigned to a team?

Reply:
For this implementation, the k variable represents the index of the student in the class list. Therefore, I only increment the index by 1 only if I added a student, else I would run another iteration of the while loop on the same student, but this time generating another team to add into.

Reply:
ah got it my bad i didnt see the k increment in the if statement 👍